test.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import signal
+from functools import partial
+from threading import Lock, Thread
+import time
+from typing import Callable
+
+import pigpio
+from luma.emulator.device import asciiblock, pygame as luma_pygame
+from PIL import ImageFont
+import pygame
+
+from display import ScreenDisplay, Terminal
+from sensor_main import Device
+
+def main():
+    logging.basicConfig(
+        format='%(asctime)s %(levelname)-8s %(message)s',
+        level=logging.DEBUG,
+        datefmt='%Y-%m-%d %H:%M:%S',
+        filename='sensor.log',
+        encoding='utf-8',
+    )
+    print(f"{pygame.init()=}")
+    emulator = PygameEmulator(320, 240, rotate=0)
+    keyboard = KeyboardMock()
+    device = Device(
+        config_file="sensors_config.ini",
+        pi_gpio=PigpioWrapper(keyboard),
+        #database=Database(),
+        display=ScreenDisplay(
+            terminal=Terminal(
+                device=emulator,
+                font=ImageFont.truetype("DejaVuSansMono.ttf", 21)
+            )
+        )
+    )
+    stop = False
+    def sigint_handler(_1, _2):
+        nonlocal stop
```

Comment:
Is `nonlocal` necessary?
Please consider.

Reply:
Yes. Without nonlocal signal handler would create local variable stop with value of True